Fixed #9.

astrohart opened the pull request:
Made the generated `Manifest` class `public`.
Made the generated `Manifest` class not be named `Vsix` but instead, `{FileNamespace}Manifest`.

This is to avoid name collisions for projects such as `ExtensionPackTools` where we have actually three different VSIXs in the same solution and then there are name collisions in shared code.

astrohart committed:
Install the Microsoft.VisualStudio.AppDesigner and Microsoft.VisualStudio.ManagedInterfaces NuGet packages

astrohart committed:
Update VsixManifestGenerator.cs

* Made the generated class public
* Made the generated class be named {FileNamespace}Manifest.

P.S. This should be user configurable; right now it's hard coded.

madskristensen commented on src/VsixSynchronizer.csproj:
```diff
@@ -84,6 +84,12 @@
     </VSCTCompile>
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.VisualStudio.AppDesigner">
```
Where are these consumed?

astrohart committed:
Update VsixManifestGenerator.cs

* Make the Manifest class have initialize-once-and-store read-only properties instead of constants, and make it a Singleton (gasp!).  This is to use a Strategy Pattern in a solution where there is more than one .vsix, among the shared code, to figure out which manifest we want to use.

These really should also implement an interface, but I am not sure how to do that only once, instead of each time a VSIX manifest is generated.

astrohart committed:
Update the generators

Made the text between the <auto-generated> tags adhere to conventions.

madskristensen commented on src/Generators/VsixManifestGenerator.cs:
```diff
@@ -36,13 +36,15 @@ private string GenerateClass(VsixManifest manifest)
+            sb.AppendLine($"        public static {FileNamespace}Manifest Instance {{ get; }} = new {FileNamespace}Manifest();");
+            sb.AppendLine();
+            sb.AppendLine($"        public string Id {{ get; }} = \"{EscapeString(manifest.ID)}\";");
+            sb.AppendLine($"        public string Name {{ get; }} = \"{EscapeString(manifest.Name)}\";");
```
The reason these are constants is such that they can be used in attributes. Why change it?

astrohart replied:
They can be left as constants after all, my bad. I was trying to make the generator produce an interface, but it's not necessary to.  I'll undo the changes here.

astrohart replied:
I've made two new commits.

astrohart committed:
Removed the NuGet packages I added in a prior commit that proved to be not needed.

astrohart committed:
Changed properties back to constants

astrohart commented:
I cloned this repo before you merged PR #5, is that why appveyor is failing?  It builds on my machine.

astrohart commented:
I am going to delete my forked repo, re-fork, re-branch, re-clone, and then re-do this PR.
